src/io/linux.zig

```diff
@@ -1438,7 +1434,7 @@ pub const IO = struct {
 
         while (true) {
             const res = stdx.fstatfs(dir_fd, &statfs);
-            switch (os.linux.getErrno(res)) {
```

Comment:
I belive this is wrong. The posix `errno` either looks at the libc threadlocal, or does the normal integer logic. If this code ends up being used when we also link libc, I think what would happen is that it'll use thread-local errno, which won't actually be set as we are doing syscalls directly here. 

I think this needs to be `os.linux.E.init`

Also: libc errno is just such a terrible interface, especially on Linux. Syscall returns you a perfectly fine error code, and then you go and stuff those exquisite bits into a thread local! Barbarity! 

Luckily, we don't use libc :o)

Reply:
I am not sure where we should be using `posix.errno` on Darwin. I think yes, because we do link libc on Darwin because there's no other way?